src/Compute/Compute/Generated/GalleryImageVersion/GalleryImageVersionCreateOrUpdateMethod.cs

```diff
@@ -260,6 +260,43 @@ public override void ExecuteCmdlet()
                         }
                     }
 
+                    if (this.IsParameterBound(c => c.SourceImageVMId))
+                    {
+                        if (!this.SourceImageVMId.ToLower().Contains("/virtualmachines/"))
+                        {
+                            throw new Exception("The `SourceImageVMId` parameter must be a valid virtual machine reference id such as /subscriptions/<sub>/resourceGroups/<rg>/providers/Microsoft.Compute/virtualMachines/<vm>");
```

Comment:
could this type of error message come from the service level?

Reply:
Not before this PS release. This error will be added in the backend when they can.